static/src/javascripts/projects/common/utils/background.js

```diff
@@ -0,0 +1,21 @@
+/*
+    runs each block on a separate animation frame to prevent excessive lumpiness
+ */
+define([
+    'common/utils/_'
+], function (
+    _
+) {
+
+    return function (codeBlocks) {
+        return (_.reduceRight(codeBlocks, function (restFunctions, fn) {
+            return function () {
+                requestAnimationFrame(function () {
+                    fn();
+                    restFunctions();
```

Comment:
You can guarantee the next chunk will be executed by reversing these: queue the next animation frame, then execute the current chunk.